File: app/src/main/java/com/example/pcmall/service/LoginRegisterService.java
package com.example.pcmall.service;

import com.example.pcmall.model.LoginUser;
import com.example.pcmall.model.response.ResponseResult;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface LoginRegisterService {
    @FormUrlEncoded
    @POST("login")
    Observable<ResponseResult<LoginUser>> login(
            @Field("uid") String uid,
            @Field("password") String password,
            @Field("code") String code);

    @FormUrlEncoded
    @POST("register")
    Observable<ResponseResult<String>> register(@Field("uid") String uid,
                                                @Field("password") String password,
                                                @Field("code") String code);

    @GET("captcha.jpg")
    Observable<ResponseResult<String>> getCaptcha();
}

File: app/src/main/java/com/example/pcmallcompose/service/LoginRegisterService.kt
package com.example.pcmallcompose.service

import com.example.pcmallcompose.model.User
import com.example.pcmallcompose.model.response.ResponseResult
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST

interface LoginRegisterService {
    @FormUrlEncoded
    @POST("login")
    suspend fun login(
        @Field("uid") uid: String,
        @Field("password") password: String,
        @Field("code") code: String
    ): ResponseResult<User>

    @FormUrlEncoded
    @POST("register")
    suspend fun register(
        @Field("uid") uid: String,
        @Field("password") password: String,
        @Field("code") code: String
    ): ResponseResult<String>

    @GET("captcha.jpg")
    suspend fun getCaptcha(): ResponseResult<String>
}